doofinder/includes/api/class-store-api.php

```diff
@@ -65,74 +76,63 @@ public function __construct()
+     *        'api_user' => 'pedro'
+     *    ),
+     *    'search_engines' => array(
+     *        'fde92a8f364b8d769262974e95d82dba' => array(
+     *          'feed_type' => 'post',
+     *          'url' => 'http://pedro-wordpress.ngrok.doofinder.com'
+     *        )
+     *    )
+     * )
+     * @return void
+     */
+    public function normalize_store_and_indices()
+    {
+        $wizard = Setup_Wizard::instance();
+        $api_keys = Setup_Wizard::are_api_keys_present($wizard->process_all_languages, $wizard->language);
+        $store_payload = $this->build_store_payload($api_keys);
 
-            foreach ($api_keys as $item) {
-                if ($item['hash'] === 'no-hash') {
-                    //Prioritize the locale code
-                    $code = $item['lang']['locale'] ?? $item['lang']['code'] ?? $primary_language;
-                    $code = $this->format_language_code($code);
-                    // Prepare search engine body
-                    $this->log->log('Wizard Step 2 - Prepare Search Enginge body : ');
-
-                    $store_data["search_engines"][] = [
-                        'name' => $domain . ($code ? ' (' . strtoupper($code) . ')' : ''),
-                        'language' => $code,
-                        'currency' => $currency,
-                        'site_url' => get_bloginfo('url'),
-                        'datatypes' => [
-                            [
-                                "name" => "post",
-                                "preset" => "generic",
-                                "datasources" => [
-                                    [
-                                        "type" => "wordpress",
-                                        "options" => [
-                                            "feed_type" => "post",
-                                            "url" =>  get_bloginfo('url')
-                                        ]
-                                    ]
-                                ]
-                            ],
-                            $product_datatype
-                        ]
-                    ];
-                }
-            }
+        $payload = [
+            'store_options' => $store_payload['options']
+        ];
 
-            $this->log->log("store_data: ");
-            $this->log->log($store_data);
+        foreach ($store_payload['search_engines'] as $search_engine) {
+            $lang = Helpers::get_language_from_locale($search_engine['language']);
 
-            return $this->sendRequest("plugins/create-store", $store_data);
+            //If the installation is not multilanguage, replace the lang with ''
+            if (is_a($this->language, No_Language_Plugin::class)) {
+                $lang = '';
+            }
+
+            $se_hashid = Settings::get_search_engine_hash($lang);
+            $payload['search_engines'][$se_hashid] = $search_engine['datatypes'][0]['datasources'][0]['options'];
```

Comment:
This is a little risky IMO. How do we know it's always the first occurrence?

Reply:
Here we need to obtain the datasource options, we access to the first occurrence because it should always return at least one search engine with one datatype and one datasource.

The rest of searchengines is not relevant for the purpose of this function.